packages/website/docs/createRecentSearchesPlugin.md

```diff
@@ -24,9 +52,45 @@ autocomplete({
 });
 ```
 
-With [Query Suggestions](createQuerySuggestionsPlugin):
+For example, you can plug it to a MongoDB database using [mongoose](https://mongoosejs.com/).
 
-```ts
+```js
+import mongoose from 'mongoose';
+import { createRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+import { search } from '@algolia/autocomplete-plugin-recent-searches/usecases/localStorage';
+
+mongoose.connect('mongodb://localhost:27017/myapp', { useNewUrlParser: true });
+
+const schema = new mongoose.Schema({
+  objectID: String,
+  query: String,
+  category: {
+    type: String,
+    default: undefined,
+  },
+});
+const RecentSearchesItem = mongoose.model('RecentSearchesItem', schema);
+
+const recentSearchesPlugin = createRecentSearchesPlugin({
+  storage: {
+    async getAll(query) {
+      const items = await RecentSearchesItem.find({});
+
+      return search({ query, items, limit: 5 });
+    },
+    onAdd(item) {
+      RecentSearchesItem.create(item);
+    },
+    onRemove(objectID) {
+      RecentSearchesItem.deleteOne({ objectID });
+    },
+  },
+});
```

Comment:
I don't have a MongoDB account so I didn't test this one out, only followed the documentation carefully.

The idea is to show an basic implementation of the storage so that users can either start from it or use it as a reference to build their own.

I used the `search` function from `./usecases/localStorage` because it abstracts the highlighting logic and because you can't use a predicate in `Model.find`, so the filtering logic from `search` is useful.

Maybe this would make sense to move this function into a more generic namespace, such as `./usecases/static`. Also, it may be useful to expose the `highlight` function as `getAll` is supposed to return an array of highlighted items, and that highlighting logic would be painful to re-implement.

Reply:
Very cool!

We're not supposed to import from `./usecases/localStorage` so if it's useful, I'll make sure to export it from the package (@redox also complained about this not being provided).

I'll take care of updating the guide once it's done.